jenkins/performance-search5-elasticsearch.groovy

```diff
@@ -1,136 +1,73 @@
 					mvn clean install -P search5 \
 					-U -pl jmh-elasticsearch -am \
 					-DskipTests -Ddocker.skip -Dtest.elasticsearch.run.skip=true \
 			"""
-				dir ('jmh-elasticsearch/target') {
-					stash name:'jar', includes:'benchmarks.jar'
-				}
-			}
-		}
+                dir ('jmh-elasticsearch/target') {
+                    stash name:'jar', includes:'benchmarks.jar'
+                }
+            }
+        }
+        stage('Performance test') {
+            steps {
+                unstash name: 'jar'
+                sh 'docker run --name postgresql -p 5431:5432 -e POSTGRES_USER=username -e POSTGRES_PASSWORD=password -e POSTGRES_DB=database -d postgres:10.5'
+                sleep(time:10,unit:"SECONDS") // wait for postgres to be ready
+                sh 'mkdir -p output'
 
-		stage ('Performance test') {
-			def awsCredentialsId = helper.configuration.file?.aws?.credentials
-			if (!awsCredentialsId) {
-				throw new IllegalStateException("Missing AWS credentials")
-			}
-			withCredentials([[$class: 'AmazonWebServicesCredentialsBinding',
-							  credentialsId   : awsCredentialsId,
-							  usernameVariable: 'AWS_ACCESS_KEY_ID',
-							  passwordVariable: 'AWS_SECRET_ACCESS_KEY'
-			]]) {
-				helper.withMavenWorkspace { // Mainly to set the default JDK
-					unstash name:'jar'
-					sh 'docker stop postgresql || true && docker rm -f postgresql || true' // stop and remove any old container
-					sh 'docker run --name postgresql -p 5431:5432 -e POSTGRES_USER=username -e POSTGRES_PASSWORD=password -e POSTGRES_DB=database -d postgres:10.5'
-					sleep(time:10,unit:"SECONDS") // wait for postgres to be ready
-					sh 'mkdir output'
-					sh """ \
+                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding',
+                                  credentialsId   : 'JenkinsSlaveServicesConsumer.amazonaws.com',
+                                  usernameVariable: 'AWS_ACCESS_KEY_ID',
+                                  passwordVariable: 'AWS_SECRET_ACCESS_KEY'
+                                 ]]) {
+                    lock('es-aws-56') {
+                        sh """ \
 							java \
 							-jar benchmarks.jar \
-							-jvmArgsAppend -Dhibernate.search.default.elasticsearch.host=$esAwsBuildEnv.endpointUris \
+							-jvmArgsAppend -Dhibernate.search.default.elasticsearch.host=$ES_AWS_56_ENDPOINT \
 							-jvmArgsAppend -Dhibernate.search.default.elasticsearch.required_index_status=yellow \
 							-jvmArgsAppend -Dhibernate.search.default.elasticsearch.aws.signing.enabled=true \
-							-jvmArgsAppend -Dhibernate.search.default.elasticsearch.aws.region=$esAwsBuildEnv.awsRegion \
+							-jvmArgsAppend -Dhibernate.search.default.elasticsearch.aws.region=$ES_AWS_REGION \
 							-jvmArgsAppend -Dhibernate.search.default.elasticsearch.aws.access_key=$AWS_ACCESS_KEY_ID \
 							-jvmArgsAppend -Dhibernate.search.default.elasticsearch.aws.secret_key=$AWS_SECRET_ACCESS_KEY \
-							-wi 1 -i 3 \
+							-wi 0 -i 2 \
```

Comment:
Why is this down to 0 warmup iterations and 2 measurement iterations?

If we want less score errors, we need:

* some warmup
* as many iterations as we can
* iterations that are as long as possible

Please try to adjust these settings to get a better score error.

Reply:
I added this commit only to make the execution of the performance tests shorter. Let's start by removing it and arrange them in the Hibernate organization repository.